SharedSources/MediaLibraryService.swift

```diff
@@ -413,8 +413,11 @@ extension MediaLibraryService: VLCMediaFileDiscovererDelegate {
         } catch let error {
             assertionFailure("MediaLibraryService: VLCMediaFileDiscovererDelegate: \(error.localizedDescription)")
         }
-
-        reload()
+        if let media = fetchMedia(with: excludeURL) {
+            requestThumbnail(for: [media])
```

Comment:
Are you sure this works? I believe that since the media haven't been added yet, `fetchMedia` will mostly return nil.

This is one of the reason of existence of the patch I sent you privately.